panc/src/main/scripts/panlint/panlint.py

```diff
@@ -0,0 +1,404 @@
+    result = []
+    annotations = RE_ANNOTATION.finditer(text)
+    for annotation in annotations:
+        start_char, end_char = annotation.span()
+        start_line = text[:start_char].count('\n') + 1
+        end_line = start_line + text[start_char:end_char].count('\n')
+        for i in range(start_line, end_line + 1):
+            result.append(i)
+    return result
+
+
+def strip_trailing_comments(line, string_ranges):
+    for comment in RE_COMMENT.finditer(line):
+        # Does the candidate comment start inside a string?
+        # If so, it's not really a comment.
+        if not inside_string(comment.start(), comment.start()+1, string_ranges):
+            debug_range(comment.start(), comment.end(), 'Comment', False)
+            line = line[:comment.start()]
+    return line.rstrip()
+
+
+def check_line_component_use(line, components_included):
+    diagnoses = []
+    messages = []
+    problem_count = 0
+
+    for m in RE_COMPONENT_USE.finditer(line):
+        if m.group('name') not in components_included:
+            message = 'Component %s in use, but component config has not been included' % m.group('name')
+            diagnoses.append(diagnose(*m.span('name')))
+            messages.append(message)
+            problem_count += 1
+
+    return diagnoses, messages, problem_count
+
+
+def check_line_patterns(line, string_ranges):
+    diagnoses = []
+    messages = []
+    problem_count = 0
+
+    for message, pattern in LINE_PATTERNS.iteritems():
+        m = pattern.search(line)
+        if m and m.group('error'):
+            start, end = m.span('error')
+            debug_range(start, end, 'Match', True)
+            if not inside_string(start, end, string_ranges):
+                diagnoses.append(diagnose(start, end))
+                messages.append(message)
+                problem_count += 1
+
+    return diagnoses, messages, problem_count
+
+
+def check_line_methods(line, string_ranges):
+    diagnoses = []
+    messages = []
+    problem_count = 0
+
+    for _, check_method in getmembers(LineChecks(), predicate=ismethod):
+        passed, diagnosis, message = check_method(line, string_ranges)
+        if not passed:
+            diagnoses.append(diagnosis)
+            messages.append(message)
+            problem_count += 1
+
+    return diagnoses, messages, problem_count
+
+
+def lint_line(line, line_number, components_included, first_line=False):
+    debug_line(line, line_number)
+
+    messages = []
+    diagnoses = []
+    problem_count = 0
+
+    if first_line:
+        first_line = False
+        if not RE_FIRST_LINE.match(line):
+            messages.append('First non-comment line must be the template type and name')
+            diagnoses.append(diagnose(0, len(line)))
+            problem_count += 1
+
+    else:
+        string_ranges = get_string_ranges(line)
+        line = strip_trailing_comments(line, string_ranges)
+
+        d, m, p = check_line_component_use(line, components_included)
```

Comment:
something tells me this will be extended in the future, so might do something like 
```python
result = [diagnoses, messages, problem_count]
def _do_check(fn, *args):
    map(lambda x,y: x+y, result, fn(*args))
_do_check(check_line_component_use, line, components_included)
...
```
not tested, but easily extenable and no copy/paste